### PR TITLE
chore: Drop deprecated persistent-fsm after deprecation cycle

### PR DESCRIPTION
**Pull request by johanandren**
Needs mention in migration guide once #32056 is merged

### CODE REVIEW

**Review by patriknw — APPROVED** [review]
LGTM, after migration guide

**Comment by johanandren** [comment]
I think i dropped too much here. I think we need to keep the adapters in typed persistence or else migrated apps will break.

**Comment by johanandren** [comment]
Making a migrated system work means pulling back a lot of the FSM stuff in addition to that adapter though, classes for events, snapshot, serialization.

File: akka-docs/src/main/paradox/project/migration-guide-2.8.x-2.9.x.md [hunk 1]
```diff
@@ -36,3 +36,8 @@ group=com.typesafe
 artifact=ssl-config-core_$scala.binary.version$
 version="0.6.1"
 }
+
+## Persistent FSM
+
+Persistent FSM has been deprecated since Akka 2.6.0 (2019-11-06) and has now been dropped. Details on how to migrate can be found in the Akka 2.7
```
Comment (patriknw) [inline akka-docs/src/main/paradox/project/migration-guide-2.8.x-2.9.x.md:42]: Akka 2.8?
The link is to 2.8

**Review by patriknw — APPROVED** [review]
LGTM, aside from...

**Pull request merged**